supervision/dataset/formats/coco.py

```diff
@@ -105,31 +135,36 @@ def detections_to_coco_annotations(
     max_image_area_percentage: float = 1.0,
     approximation_percentage: float = 0.75,
 ) -> Tuple[List[Dict], int]:
-    coco_annotations = []
+    annotation = []
     for xyxy, mask, _, class_id, _, _ in detections:
-        box_width, box_height = xyxy[2] - xyxy[0], xyxy[3] - xyxy[1]
-        polygon = []
         if mask is not None:
-            polygon = list(
-                approximate_mask_with_polygons(
-                    mask=mask,
-                    min_image_area_percentage=min_image_area_percentage,
-                    max_image_area_percentage=max_image_area_percentage,
-                    approximation_percentage=approximation_percentage,
-                )[0].flatten()
+            polygons = approximate_mask_with_polygons(
+                mask=mask,
+                min_image_area_percentage=min_image_area_percentage,
+                max_image_area_percentage=max_image_area_percentage,
+                approximation_percentage=approximation_percentage,
             )
-        coco_annotation = {
-            "id": annotation_id,
-            "image_id": image_id,
-            "category_id": int(class_id),
-            "bbox": [xyxy[0], xyxy[1], box_width, box_height],
-            "area": box_width * box_height,
-            "segmentation": [polygon] if polygon else [],
-            "iscrowd": 0,
-        }
-        coco_annotations.append(coco_annotation)
-        annotation_id += 1
-    return coco_annotations, annotation_id
+            for polygon in polygons:
+                if polygon.any():  # polygon not empty
+                    next_object = object_to_coco(
+                        xyxy=xyxy,
```

Comment:
That's not the correct box; you save each polygon as a separate entry but still use the original box. We need to calculate a new box for each polygon. I recommend using the `polygon_to_xyxy` function.

![IMG_6F79B4D94BBA-1](https://github.com/roboflow/supervision/assets/26109316/556a0e90-5260-4258-8439-48f57976b5eb)

Reply:
@Youho99 It looks like the COCO format supports disjointed masks - you can have multiple masks representing the same object. I think we should go for this.

https://github.com/roboflow/supervision/issues/373#issuecomment-1831781340

Reply:
I agree in theory with this.

Now, doing the implementation the way I did it, I get exactly the same number of polygons as using as_yolo()

So if we define that the method of calculating polygons should change, then it would be better to also look at how this is done in as_yolo() and as_pascal_voc().

Reply:
The question is this:

Do we want the results of a conversion to coco, yolo and pascal voc to have exactly the same result (in the format I mean)?

Because it does not seem to me that it is possible to manage disjointed masks in yolo for example (except by applying a technique allowing the masks to be connected by thin strips)

Reply:
Example : https://github.com/roboflow/supervision/issues/574

Reply:
So I don't know exactly what to start with for this part.

I'm waiting for your feedback on this.

Reply:
Welcome back after the weekend, @Youho99. 👋🏻 For now, don't worry about the consistency of the `as_coco`, `as_yolo`, and `as_pascal_voc` methods. If necessary, we'll open a new task to address the differences.

Converting a disjointed mask into several separate detections is an undesirable result of the lack of support for such annotations in YOLO and PASCAL formats. I believe that if it's possible to avoid such conversions in the COCO format, we should do so.

Reply:
Ok, I'll look that way then

Note that according to my tests on the yolo conversion, during mask disjoint, it only saves the first mask (for as_yolo())